File: app/src/main/java/com/tangem/data/network/BlockchainInfoApi.java
package com.tangem.data.network;

import com.tangem.data.network.model.BlockchainInfoAddress;
import com.tangem.data.network.model.BlockchainInfoUnspents;

import io.reactivex.Single;
import okhttp3.ResponseBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface BlockchainInfoApi {
    @GET(Server.ApiBlockchainInfo.Method.ADDRESS)
    Single<BlockchainInfoAddress> blockchainInfoAddress(@Path("address") String address, @Query("offset") Integer offset);

    @GET(Server.ApiBlockchainInfo.Method.UTXO)
    Single<BlockchainInfoUnspents> blockchainInfoUnspents(@Query("active") String address);

    @FormUrlEncoded
    @POST(Server.ApiBlockchainInfo.Method.PUSH)
    Single<ResponseBody> blockchainInfoPush(@Field("tx") String tx);
}

File: blockchain/src/main/java/com/tangem/blockchain/blockchains/bitcoin/network/blockchaininfo/BlockchainInfoApi.kt
package com.tangem.blockchain.blockchains.bitcoin.network.blockchaininfo

import okhttp3.ResponseBody
import retrofit2.http.*

interface BlockchainInfoApi {
    @GET("rawaddr/{address}?limit=5")
    suspend fun getAddress(@Path("address") address: String): BlockchainInfoAddress

    @GET("unspent")
    suspend fun getUnspents(@Query("active") address: String): BlockchainInfoUnspents

    @FormUrlEncoded
    @POST("pushtx")
    suspend fun sendTransaction(@Field("tx") transaction: String): ResponseBody
}
